tools/functional-tester/etcd-runner/main.go

```diff
@@ -310,6 +318,23 @@ func runWatcher(eps []string) {
 	wg.Wait()
 }
 
+func getModRevForKey(client *clientv3.Client, ctx context.Context, key string) int64 {
+	var modrevision int64 = 0
+	for {
+		gr, err := client.Get(ctx, key)
```

Comment:
this part looks similar to line 245 -> 250. probably add a func called getWithRetry or something?